src/test/fixtures/utils.js

```diff
@@ -58,3 +59,24 @@ export function formatTree(
 
   return lines;
 }
+
+/**
+ * Wait until the Redux store gets into a specific state given a predicate function.
+ * Generally prefer hooking into existing promises, but this can be used in cases,
+ * like when clicking a component link in a test, where there is no Promise to
+ * wait on.
+ */
+export function waitUntilState(
+  store: Store,
+  predicate: State => boolean
+): Promise<void> {
+  return new Promise(resolve => {
+    store.subscribe(() => {
+      if (predicate(store.getState())) {
+        // Resolve the next Promise tick, and allow all other store subscribers
+        // to update first.
+        new Promise(() => resolve());
```

Comment:
I think you meant something like:
 `Promise.resolve().then(resolve)`

When calling `new Promise(promiseCreator)` the `promiseCreator` is called synchronously.

Reply:
Ah good point, now I don't know why this is working with it wrong. Still, I'll make this change.